src/util/createServerRootMixin.js

```diff
@@ -15,26 +15,28 @@ function walkIndex(indexWidget, visit) {
   });
 }
 
-export function renderToString(app) {
-  let _renderToString;
-  try {
-    _renderToString = isVue3
-      ? require('@vue/server-renderer').renderToString
-      : require('vue-server-renderer/basic');
-  } catch (e) {
-    // error is handled by regular if, in case it's `undefined`
-  }
-  if (!_renderToString) {
-    if (isVue3) {
+export async function renderToString(app) {
+  if (isVue3) {
+    let module;
+    try {
+      module = await import('@vue/server-renderer');
+    } catch (err) {
+      // error is handled by regular if, in case it's `undefined`
+    }
+    if (!module) {
       throw new Error('you need to install @vue/server-renderer');
-    } else {
-      throw new Error('you need to install vue-server-renderer');
```

Comment:
why is the vue 2 code removed?

wasn't this in specific files at some point?

Reply:
It's still below down there.